lib/spack/spack/cmd/containerize.py

```diff
@@ -47,6 +52,9 @@ def containerize(parser, args):
 
     config = spack.container.validate(config_file)
 
+    if args.os:
+        config['spack']['container']['images']['os'] = args.os
```

Comment:
I think we need to:
1. Validate the `args.os` argument to ensure it's among the allowed values
2. Handle the case where a conflicting option is already present in the `spack.yaml` and e.g. inform the user about it

For 2. in particular, there are currently different ways to select the build image. If a `spack.yaml` already defines  [custom images](https://spack.readthedocs.io/en/latest/containers.html#use-custom-images-provided-by-users) we should probably error and report a comprehensible message pointing to the docs.

Reply:
@philregier Apologies, since I remember we briefly discussed that a few weeks ago during the meeting, but can you remind me which use case is covered by this PR?

Reply:
Sorry, @alalazo, I was getting ready for travel when you made these comments and I didn't see them when I got back; that was my fault.  To your comments:

1.  Due to the placement, `args.os` gets treated exactly like text in `spack.yaml` does; if I specify `whatever` in either location, I get the same error:
```
==> Error: invalid operating system name "whatever". [Allowed values are alpine:3, amazonlinux:2, centos:stream, centos:7, opensuse/leap:15, nvidia/cuda:11.2.1, ubuntu:22.04, ubuntu:20.04, ubuntu:18.04]
```
2.  I based my suggestion on the [documentation](https://spack.readthedocs.io/en/latest/configuration.html#configuration-scopes), and am proposing this as a level-7 override to the level-6 `spack.yaml` (see below for use cases).  I don't really get what this has to do with custom images, but if having an entry in spack:container:images:os would presently error and report a comprehensible message pointing to the docs, then specifying `--os` probably will as well, unless aforementioned error takes place in `spack.container.validate`, in which case I would definitely need to revisit my proposal.

As for use cases, the basic ones include (1) troubleshooting:  if a spack.yaml _with_ a defined `os` runs into problems, one could override it on the command line to see if the recipe produced with an alternate `--os` override had the same  issue, (2) automation, so a distribution vendor can do CI/CD in a parameterized fashion, testing the same `spack.yaml` against multiple targets (without adding another dimension to the required matrix of spack.yaml files) to make sure all OS releases are suitable for use as spack targets, and (3) for documentation purposes, to simplify demonstration of exporting the same functioning spack environment to different target containers.

This last gets to my other concern, which isn't quite covered by your questions so far, which is that `containerize` strikes me very much as an `operation` on an environment, not a fixed set of `properties`.  I don't see any reason why if Alice, Bob, and Charlie are using the same environment on their institutional cluster, why Alice might not be perfectly content to run locally, Bob might want to export container images from their environment to Amazon, and Charlie might want to build CentOS images for use on GCP.  I don't get why the final export step should mean they have to all have forks of their shared environment, since (again) the `containerize` step seems to be an _export operation_ as opposed to a property of the instantiated environment.

Does any of this make sense?  I don't presently want to change what spack does in any of these cases, just add a switch to access its current unmodified behavior from an automated and/or parameterized context.